### ISSUE TITLE
Does this still work on latest builds?

### ISSUE THREAD

**Issue by smadger** [m-i-k-e-e/WifiPortalAutoLog#1]
When I install this on either a Samsung J5 (running Android 7) or a Pixel 2 (running Android 8), I do not get a prompt to select the app when I connect to a captive portal and tap on the Sign In To Wi-Fi notification. It always uses the system default app to sign in.

Does it still work for you?

I'm building this in Android Studio 3.1 (don't know if that matters)

**Comment by m-i-k-e-e** [comment]
Yep, it still works on 8.1. You might have set the default connection to 'always use'.I must also mention that the broadcast listener branch is buggy.Regards,MichaelLe 10 avr. 2018 14:32, Ian Smith <notifications@github.com> a écrit :When I install this on either a Samsung J5 (running Android 7) or a Pixel 2 (running Android 8), I do not get a prompt to select the app when I connect to a captive portal and tap on the Sign In To Wi-Fi notification. It always uses the system default app to sign in.
Does it still work for you?
I'm building this in Android Studio 3.1 (don't know if that matters)

**Comment by smadger** [comment]
Thanks Michael. That was the problem on the Pixel - the CaptivePortalLogin app has it's 'Open By Default' set and clearing it fixed the issue.

But there was no default set on the Samsung J5 and it doesn't give the option to use the WifiPortalAutoLog app when tapping on the Sign-in notification, even after a factory reset. It just uses the native app.

Same behaviour for a Samsung S8.

I was thinking it was just a Samsung thing, but I tried it on an old HTC One M9 running Android 6.0 and it has the same behaviour there :-(

**Comment by m-i-k-e-e** [comment]
Weird, I have a A5 at home, I'll try.Le 10 avr. 2018 17:29, Ian Smith <notifications@github.com> a écrit :Thanks Michael. That was the problem on the Pixel - the CaptivePortalLogin app has it's 'Open By Default' set and clearing it fixed the issue.
But there was no default set on the Samsung J5 and it doesn't give the option to use the WifiPortalAutoLog app when tapping on the Sign-in notification, even after a factory reset. It just uses the native app.
Same behaviour for a Samsung S8.
I was thinking it was just a Samsung thing, but I tried it on an old HTC One M9 running Android 6.0 and it has the same behaviour there :-(

**Comment by m-i-k-e-e** [comment]
I just tried on my tab S2 and same issue. I'll try to find a solution (if it exists). Maybe the broadcast branch can help

**Comment by smadger** [comment]
Looking at the full logs on a Samsung device, you can see this when it detects a captive portal:

04-11 14:33:49.068 2978-25888/? I/ActivityManager: START u0 {act=android.intent.action.SEND typ=null flg=0x10c00000 cmp=ComponentInfo{com.android.captiveportallogin/com.android.captiveportallogin.CaptivePortalLoginActivity}} from uid 1000 on display 0


Note the SEND action, rather than the CAPTIVE_PORTAL one.

Compare it with this from the Pixel:

04-11 15:43:32.941 1145-3971/? I/ActivityManager: START u0 {act=android.net.conn.CAPTIVE_PORTAL flg=0x10400000 cmp=android/com.android.internal.app.ResolverActivity (has extras)} from uid 1000

Using an app called ManifestViewer I can see that on Samsung, the manifest for CaptivePortalLogin has this as the intent-filter:

            <intent-filter>
                <action android:name="android.net.conn.CAPTIVE_PORTAL" />
                <action android:name="android.intent.action.ACTION_SEND" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
			
It seems like Samsung are overriding the default method of launching the CaptivePortalLogin and are using an explicit intent to start com.android.captiveportallogin

**Comment by shailangsa** [comment]
Hello hope you are well, I tried running this on an android 9 tablet, and on the main activity it seems to display only the username and password button which allows me to enter them but after that nothing else occurs nor can i see a submit button.

When does it log into the captive portal?

**Comment by m-i-k-e-e** [comment]
Hey @shailangsa, 

I don't know if it still works, I don't use it anymore. 

The goal of this application was to automatically log you on a captive portal, there is non need of a submit button, there is a system intent sent : android.net.conn.CAPTIVE_PORTAL. This intent triggers the logon process

Of course this tool was made for the portal I was using, you will need to adapt the javascript part in the main activity (line 62) to suit your need (inject username and password and press the login button). 
The first step is to get open the source code of the page and then retrieve the different id's. The portal I was using also requires me to check a box (document.getElementById('aupAccepted').click()), it might not be needed by yours.

> Hello hope you are well, I tried running this on an android 9 tablet, and on the main activity it seems to display only the username and password button which allows me to enter them but after that nothing else occurs nor can i see a submit button.
> 
> When does it log into the captive portal?

**Comment by shailangsa** [comment]
oh right, but if  incorrect javascript is used, would it still show an empty activity as its showing to me right now?

Also as a side note, i think an alternative way to login to a captive portal in app would be to visit a non ssl URL in a webview inside the app , then it would forward the user to the captive portal page?

**Comment by m-i-k-e-e** [comment]
The screen is not empty, there is a hidden webview which is used to load the portal and a textview to show the current status.

I don't see the point of forwarding the user to the portal, the system already does it. The goal was to log you in every time you connect to the portal. with no interaction. The view is supposed to close itself once logged in

**Comment by shailangsa** [comment]
ok, i should have been more clear. i meant to ensure that the user (which will be using an app on a device that will be provided to them where the app will be in screen pinned mode so they wouldnt actually get to access the default wifi settings) logs into the captive portal, would be to have the app open a non ssl URL in the in-app webview which would automatically forward them to a captive portal page?
